homeassistant/components/media_player/dlna_dmr.py

```diff
@@ -0,0 +1,542 @@
+    MediaPlayerDevice,
+    PLATFORM_SCHEMA)
+from homeassistant.const import (
+    EVENT_HOMEASSISTANT_STOP,
+    CONF_URL, CONF_NAME,
+    STATE_OFF, STATE_ON, STATE_IDLE, STATE_PLAYING, STATE_PAUSED)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+
+DOMAIN = 'dlna_dmr'
+
+REQUIREMENTS = [
+    'async-upnp-client==0.12.0',
+]
+
+DEPENDENCIES = ['http']
+
+DEFAULT_NAME = 'DLNA Digital Media Renderer'
+DEFAULT_LISTEN_PORT = 8301
+
+CONF_LISTEN_IP = 'listen_ip'
+CONF_LISTEN_PORT = 'listen_port'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL): cv.string,
+    vol.Optional(CONF_LISTEN_IP): cv.string,
+    vol.Optional(CONF_LISTEN_PORT, default=DEFAULT_LISTEN_PORT): cv.port,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+SUPPORT_DLNA_DMR = \
+    SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_SET | \
+    SUPPORT_PLAY | SUPPORT_STOP | SUPPORT_PAUSE | \
+    SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | \
+    SUPPORT_PLAY_MEDIA
+
+HOME_ASSISTANT_UPNP_CLASS_MAPPING = {
+    'music': 'object.item.audioItem',
+    'tvshow': 'object.item.videoItem',
+    'video': 'object.item.videoItem',
+    'episode': 'object.item.videoItem',
+    'channel': 'object.item.videoItem',
+    'playlist': 'object.item.playlist',
+}
+HOME_ASSISTANT_UPNP_MIME_TYPE_MAPPING = {
+    'music': 'audio/*',
+    'tvshow': 'video/*',
+    'video': 'video/*',
+    'episode': 'video/*',
+    'channel': 'video/*',
+    'playlist': 'playlist/*',
+}
+UPNP_DEVICE_MEDIA_RENDERER = [
+    'urn:schemas-upnp-org:device:MediaRenderer:1',
+    'urn:schemas-upnp-org:device:MediaRenderer:2',
+    'urn:schemas-upnp-org:device:MediaRenderer:3',
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+SETUP_LOCK = asyncio.Lock()
+
+
+def determine_listen_ip(target_url, config):
+    """
+    Determine the IP and port to listen on.
+
+    If specified in config, use config.
+    Otherwise try to figure it out.
+    """
+    server_host = config.get(CONF_LISTEN_IP)
+    server_port = config.get(CONF_LISTEN_PORT)
+
+    if server_host is None:
+        # determine server_host by opening a UDP socket to target,
+        # but don't actually send anything
+        parsed = urllib.parse.urlparse(target_url)
+        target_host = parsed.hostname
+        target_port = parsed.port
+        temp_sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        temp_sock.connect((target_host, target_port))
+        server_host = temp_sock.getsockname()[0]
+        temp_sock.close()
+
+    return server_host, server_port
+
+
+async def async_start_notify_view(hass, server_host, server_port, requester):
+    """Register notify view."""
+    hass_data = hass.data[DOMAIN]
+    if 'notify_view' in hass_data:
+        return hass_data['notify_view']
+
+    # fire up a HTTP server
+    server = HomeAssistantHTTP(
```

Comment:
Why do we need to start a new server?

Don't use the `HomeAssistantHTTP` class but use the base aiohttp web server class.

See https://github.com/home-assistant/home-assistant/pull/15530.

Reply:
I've found that if hass is configured to use SSL and/or external FQDN, e.g. `base_url: https://home.yottatsa.name`, then subscription are being refused on some media players. See [comment](https://github.com/home-assistant/home-assistant/pull/14749/#issuecomment-403295181)

Reply:
To listen for notifications, as emulated_hue does.